vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewImporter.java

```diff
@@ -989,8 +989,19 @@ private void removeReferences(@NotNull Node node) throws ReferentialIntegrityExc
         VersioningState vs = new VersioningState(stack, node);
         Node updatedNode = null;
         Optional<String> identifier = ni.getIdentifier();
+
+        boolean isUuidProtected = false;
+        if (! session.getRepository().getDescriptor(Repository.REP_NAME_DESC).contains("Oak")) {
+            // workaround for non-Oak repos that may not be able to add
+            // mixin:referencable *after* setting jcr:uuid
+            isUuidProtected = true;
```

Comment:
The original idea was to encapsulate impl specific differences in https://github.com/apache/jackrabbit-filevault/blob/master/vault-core/src/main/java/org/apache/jackrabbit/vault/fs/spi/ServiceProvider.java.

Reply:
Thanks for the pointer; let's do more research on the "setting of jcr:uuid" first, and the outcome of that would inform what information we would want to add there.